PowerPointLabs/PowerPointLabs/SpeechEngine/TextToSpeech.cs

```diff
@@ -59,11 +67,105 @@ public static void SpeakString(String textToSpeak)
             PromptToAudio.Speak(builder);
         }
 
+        private static void SaveStringToWaveFileWithHumanVoice(string textToSave, string filePath)
+        {
+            string accessToken;
+            string textToSpeak = GetHumanSpeakNotesForText(textToSave);
+
+            try
+            {
+                Authentication auth = Authentication.GetInstance();
+                accessToken = auth.GetAccessToken();
+                Logger.Log("Token: " + accessToken);
+            }
+            catch (Exception ex)
+            {
+                Logger.Log("Failed authentication.");
+                Logger.Log(ex.ToString());
+                Logger.Log(ex.Message);
+                return;
+            }
+            string requestUri = UserAccount.GetInstance().GetUri();
+            var cortana = new Synthesize();
+
+            cortana.OnAudioAvailable += SaveAudioToWaveFile;
+            cortana.OnError += ErrorHandler;
+
+            // Reuse Synthesize object to minimize latency
+            cortana.Speak(CancellationToken.None, new Synthesize.InputOptions()
+            {
+                RequestUri = new Uri(requestUri),
+                Text = textToSpeak,
+                VoiceType = humanVoice.voiceType,
+                Locale = humanVoice.Locale,
+                VoiceName = humanVoice.voiceName,
+                // Service can return audio in different output format.
+                OutputFormat = AudioOutputFormat.Riff24Khz16BitMonoPcm,
+                AuthorizationToken = "Bearer " + accessToken,
+            }, filePath).Wait();

```

Comment:
Was just wondering, if the service returns an error after the async request, what kind of error would you display? 

For example, the user has already logged in but the user's internet suddenly goes down and we are unable to contact the service. Another case would be that the service suddenly starts returning errors to us. Would there be an error message to the user?  It would be good if you gave a bit more details for your solution on this PR, especially how you send an async request to the speech service and convert the response to the appropriate WAV format. Thanks!